pkg/kubelet/config/file_test.go

```diff
@@ -128,47 +135,56 @@ func TestReadFromFile(t *testing.T) {
 				},
 			}),
 		},
+	}
+
+	for _, testCase := range testCases {
+		func() {
+			file := writeTestFile(t, os.TempDir(), "test_pod_config", testCase.fileContents)
+			defer os.Remove(file.Name())
+
+			ch := make(chan interface{})
+			NewSourceFile(file.Name(), hostname, time.Millisecond, ch)
+			select {
+			case got := <-ch:
+				update := got.(kubelet.PodUpdate)
+				for _, pod := range update.Pods {
+					if errs := validation.ValidatePod(pod); len(errs) > 0 {
+						t.Errorf("%s: Invalid pod %#v, %#v", testCase.desc, pod, errs)
+					}
+				}
+				if !api.Semantic.DeepEqual(testCase.expected, update) {
+					t.Errorf("%s: Expected %#v, Got %#v", testCase.desc, testCase.expected, update)
+				}
+			case <-time.After(time.Second):
+				t.Errorf("%s: Expected update, timeout instead", testCase.desc)
+			}
+		}()
+	}
+}
+
+func TestReadPodsFromFile(t *testing.T) {
+	hostname := "random-test-hostname"
+	var testCases = []struct {
+		desc     string
+		pod      runtime.Object
+		expected kubelet.PodUpdate
+	}{
 		{
-			desc: "Manifest v1beta2",
-			fileContents: `{
-					"version": "v1beta2",
-					"uuid": "12345",
-					"id": "test",
-					"containers": [{ "name": "image", "image": "test/image", "imagePullPolicy": "PullAlways"}]
-				}`,
-			expected: CreatePodUpdate(kubelet.SET, kubelet.FileSource, &api.Pod{
+			desc: "Simple pod",
+			pod: &api.Pod{
+				TypeMeta: api.TypeMeta{
+					Kind:       "Pod",
+					APIVersion: "",
+				},
 				ObjectMeta: api.ObjectMeta{
-					Name:      "test-" + hostname,
+					Name:      "test",
 					UID:       "12345",
-					Namespace: kubelet.NamespaceDefault,
-					SelfLink:  "/api/v1beta2/pods/test-" + hostname + "?namespace=default",
+					Namespace: "mynamespace",
 				},
 				Spec: api.PodSpec{
-					Host:          hostname,
-					RestartPolicy: api.RestartPolicyAlways,
```

Comment:
Ok, I see you're applying conversion now. I'd only expect defaults to be applied when converting from a versioned representation to the internal rep.